classes/AlertView/Standard.php

```diff
@@ -0,0 +1,391 @@
+                } catch ( \MySociety\TheyWorkForYou\MemberException $e ) {
+                    // do nothing
+                }
+            }
+            $this->data['constituencies'] = $cons;
+        }
+    }
+
+    private function addAlert() {
+        $external_auth = auth_verify_with_shared_secret($this->data['email'], OPTION_AUTH_SHARED_SECRET, get_http_var('sign'));
+        if ($external_auth) {
+            $confirm = false;
+        } elseif ($this->data['email_verified']) {
+            $confirm = false;
+        } else {
+            $confirm = true;
+        }
+
+        // If this goes well, the alert will be added to the database and a confirmation email
+        // will be sent to them.
+        $success = $this->alert->add( $this->data, $confirm );
+
+        if ($success>0 && !$confirm) {
+            $result = 'alert-added';
+        } elseif ($success>0) {
+            $result = 'alert-confirmation';
+        } elseif ($success == -2) {
+            // we need to make sure we know that the person attempting to sign up
+            // for the alert has that email address to stop people trying to work
+            // out what alerts they are signed up to
+            if ( $this->data['email_verified'] || ( $this->user->loggedin && $this->user->email() == $this->data['email'] ) ) {
+                $result = 'alert-exists';
+            } else {
+                // don't throw an error message as that implies that they have already signed
+                // up for the alert but instead pretend all is normal but send an email saying
+                // that someone tried to sign them up for an existing alert
+                $result = 'alert-already-signed';
+                $this->alert->send_already_signedup_email($this->data);
+            }
+        } else {
+            $result = 'alert-fail';
+        }
+
+        // don't need these anymore so get rid of them
+        $this->data['keyword'] = '';
+        $this->data['pid'] = '';
+        $this->data['alertsearch'] = '';
+        $this->data['pc'] = '';
+
+        $this->data['results'] = $result;
+        $this->data['criteria'] = $this->prettifyCriteria($this->alert->criteria);
+    }
+
+
+    private function formatSearchTerms() {
+        $this->data['alertsearch_pretty'] = $this->data['alertsearch'];
+        if ( $this->data['alertsearch'] && preg_match('#speaker:(\d+)#', $this->data['alertsearch'], $m)) {
+            $MEMBER = new \MEMBER(array( 'person_id' => $m[1] ));
+            $alertsearch_pretty = str_replace("speaker:$m[1]", "speaker:" . $MEMBER->full_name(), $this->data['alertsearch']);
+            $this->data['alertsearch_pretty'] = $alertsearch_pretty;
+        }
+
+        if ( $this->data['alertsearch'] ) {
+            $this->data['search_text'] = $this->data['alertsearch'];
+        } else {
+            $this->data['search_text'] = $this->data['keyword'];
+        }
+    }
+
+    private function prettifyCriteria($alert_criteria) {
```

Comment:
The current list of your alerts spots multiple speaker: in it, this function doesn't (though I realise criteria_pretty doesn't either, just the current list of alerts didn't use that, wonder if it's way is better...).